rumqttc/src/mqttbytes/mod.rs

```diff
@@ -13,7 +13,7 @@ pub mod v4;
 pub use topic::*;
 
 /// Error during serialization and deserialization
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, thiserror::Error)]
```

Comment:
Adding `#[error("text: {0}")]` attributes above each variant, might be a good idea?

Reply:
I was too lazy to add the text everywhere and checked out how it looks:

`Error = PayloadTooLong` with the name from the code wasnt too bad. Its better to write it manually but this way its already an improvement.

In can add all the `#[error("text: {0}")]` descriptions tho as it looks nicer then.

Reply:
I plan on writing the messages tomorrow

Reply:
Not sure about all the messages. There might be a mistake somewhere as I tried to guess what it might mean sometimes.